src/js/components/TaskDetail.js

```diff
@@ -151,56 +151,119 @@ class TaskSidePanelContents extends SidePanelContents {
-              <div className="media-object-item">
-                {statusIcon}
-              </div>
+      <PageHeader
+        icon={taskIcon}
+        subTitle={TaskStates[task.state].displayName}
+        navigationTabs={tabs}
+        mediaWrapperClassName="media-object-spacing-wrapper"
+        title={task.getId()} />
+    );
+  }
 
-              <div className="media-object-item">
-                <span className={statusClassName}>
-                  {TaskStates[task.state].displayName}
-                </span>
-              </div>
+  getTaskStatus(task) {
+    if (task == null || task.status == null) {
+      return 'Unknown';
+    }
+    return task.status;
+  }
 
-              <div className="media-object-item">
-                <span className="side-panel-subheader side-panel-subheader-emphasize">
-                  {node.hostname}
-                </span>
-              </div>
-            </div>
-          </div>
-        </div>
-        <div className="container container-pod container-pod-short container-fluid flush">
-          {this.getResources(task)}
-        </div>
-      </div>
+  getTaskEndpoints(task) {
+    let service = MarathonStore.getServiceFromTaskID(task.id);
+
+    if ((task.ports == null || task.ports.length === 0) &&
+        (task.ipAddresses == null || task.ipAddresses.length === 0)) {
+      return (<dd>None</dd>);
+    }
+
+    if (service != null &&
+        service.ipAddress != null &&
+        service.ipAddress.discovery != null &&
+        service.ipAddress.discovery.ports != null &&
+        task.ipAddresses != null &&
+        task.ipAddresses.length > 0) {
+
+      let ports = service.ipAddress.discovery.ports;
+      let endpoints = task.ipAddresses.reduce(function (memo, address) {
+        ports.forEach(port => {
+          memo.push(`${address.ipAddress}:${port.number}`);
+        });
+        return memo;
+      }, []);
+
+      if (endpoints.length) {
+        return endpoints.map(function (endpoint) {
+          return (
+            <a href={`//${endpoint}`} target="_blank">{endpoint}</a>
+          );
+        });
+      }
+
+      return 'n/a';
+    }
+
+    return task.ports.map(function (port) {
+      let endpoint = `${task.host}:${port}`;
+      return (
+        <a href={`//${endpoint}`} target="_blank">{endpoint}</a>
+      );
+    });
+  }
+
+  getMarathonTaskDetailsDescriptionList() {
+    let task = MarathonStore.getTaskFromTaskID(this.props.params.taskID);
+
+    if (task == null) {
+      return null;
+    }
+
+    let headerValueMapping = {
+      Host: task.host,
+      Ports: task.ports,
+      Endpoints: this.getTaskEndpoints(task),
+      Status: this.getTaskStatus(task),
+      'Staged at': task.stagedAt,
+      'Started at': task.startedAt,
+      Version: task.version
+    };
+
+    return (
+      <DescriptionList
+        className="container container-fluid flush container-pod container-pod-super-short flush-top"
```

Comment:
As `labels` can potentially be very long, encoded strings I suggest adding a `ddClassName="text-overflow-break-word"` to constraint the width of the page. 

Here's a gif for illustration purposes:

![wordwrap](https://cloud.githubusercontent.com/assets/1078545/15147020/86232ec0-16be-11e6-9321-fec1ab8b7987.gif)